kusto/ingest/ingest.go

```diff
@@ -57,24 +57,18 @@ type Ingestion struct {
 	connMu     sync.Mutex
 	streamConn *conn.Conn
 
-	uploadBlockSize   int
-	uploadConcurrency uint16
+	bufferSize int
+	maxBuffers int
 }
 
 // Option is an optional argument to New().
 type Option func(s *Ingestion)
 
-// WithUploadBlockSize sets the block size for uploading blobs to kusto. Defaults to 8MB.
-func WithUploadBlockSize(size int) Option {
+// WithStaticBuffer sets a static buffer with a buffer size and max amount of buffers for uploading blobs to kusto.
+func WithStaticBuffer(bufferSize int, maxBuffers int) Option {
 	return func(s *Ingestion) {
-		s.uploadBlockSize = size
-	}
-}
-
-// WithUploadConcurrency sets the concurrency for uploading blobs to kusto. Defaults to 50.
-func WithUploadConcurrency(concurrency uint16) Option {
-	return func(s *Ingestion) {
-		s.uploadConcurrency = concurrency
+		s.bufferSize = bufferSize
+		s.maxBuffers = maxBuffers
```

Comment:
Do you want to enforce here or elsewhere a validation check on minimal values for *both* arguments?

Reply:
Now checking both non-zero, as we discussed